src/core/state/favorites/index.ts

```diff
@@ -37,32 +38,65 @@ export interface FavoritesState {
 }
 
 export const favoritesStore = createStore<FavoritesState>(
-  (set, get) => ({
-    favorites: getInitialFavorites(),
-    setFavorites: (favorites) => set({ favorites }),
-    addFavorite: ({ address, chainId }: UpdateFavoritesArgs) => {
-      const { favorites } = get();
-      const currentFavorites = favorites[chainId] || [];
-      set({
-        favorites: {
-          ...favorites,
-          [chainId]: [...currentFavorites, address],
+  (set, get) => {
+    const trackFavoriteChange = (
```

Comment:
is there any reason to put this function inside of the Zustand store?

Reply:
It was my understanding that it doesn't negatively impact the store and i thought i was better to centralize the logic in one place than duplicating the logic wherever we are importing the `useFavoritesStore`

Reply:
We could keep it in this file, but I'd move it outside of the store creator. Tbh might just be a nitpick from me but seems like a weird pattern.

Reply:
na its a valid nit. can move.